### ISSUE TITLE
DropdownMenu has secret focusable regions for each item in the menu

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#143507]
STEPS TO REPRODUCE

1. Run the app below.
2. Tap the "tab" key 7 times.

EXPECTED RESULTS
The focus should cycle through Before 1, Before 2, Before 3, the text field, the drop-down arrow, After, and back to Before 1.

ACTUAL RESULTS
When the focus would enter the text field, it instead goes to the values that are in the dropdown. You can see this by pressing the "enter" key, it suddenly changes the text field value to "3".

CODE FOR TEST PROGRAM
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      home: Material(
        child: Row(
          children: [
            Expanded(
              child: FilledButton(
                child: const Text('Before 1'),
                onPressed: () {},
              ),
            ),
            Expanded(
              child: FilledButton(
                child: const Text('Before 2'),
                onPressed: () {},
              ),
            ),
            Expanded(
              child: FilledButton(
                child: const Text('Before 3'),
                onPressed: () {},
              ),
            ),
            DropdownMenu<int>(
              dropdownMenuEntries: List.generate(
                5,
                growable: false,
                (int index) => DropdownMenuEntry<int>(value: index, label: '$index'),
              ),
            ),
            Expanded(
              child: FilledButton(
                child: const Text('After'),
                onPressed: () {},
              ),
            ),
          ],
        ),
      ),
    ),
  );
}
```

**Comment by sabin26** [comment]
Ref: [[DropdownMenu] Focus Traversal does not work between multiple fields](https://github.com/flutter/flutter/issues/131120)

> I just tested adding more entries in the `DropdownMenu` and concluded that **more the number of entries, more will be the number of focus traversal** to get to the `DropdownMenu` field.
> 
> So, if the `DropdownMenu` field that is going to be focused has 6 entries, it will take 6+1 focus traversal (eg: [TAB] key presses) before the field finally gets focus. This happens even if the menu is not opened.
> 
> Looks like focus goes through the entries first and then finally to the input field which is just the reverse of what should have been the correct way to focus. i.e. Focus the input field first & then the entries.

**Comment by Hixie** [comment]
Yeah, this is a dupe of that bug.
